tests/Unit/Helpers/Elliptic/FirstOrderSystem.hpp

```diff
@@ -0,0 +1,318 @@
+ * - They can be applied to a DataBox, i.e. their argument tags are consistent
+ *   with their apply function.
+ */
+template <typename System>
+void test_first_order_fluxes_computer(
+    const typename System::fluxes& fluxes_computer,
+    const DataVector& used_for_size) {
+  using FluxesComputer = typename System::fluxes;
+  static constexpr size_t volume_dim = System::volume_dim;
+  using vars_tag = typename System::fields_tag;
+  using primal_fields = typename System::primal_fields;
+  using auxiliary_fields = typename System::auxiliary_fields;
+  using VarsType = typename vars_tag::type;
+  using fluxes_tag =
+      db::add_tag_prefix<::Tags::Flux, vars_tag, tmpl::size_t<volume_dim>,
+                         Frame::Inertial>;
+  using argument_tags = typename FluxesComputer::argument_tags;
+
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> dist(0.5, 2.);
+
+  // Generate random variables
+  const auto vars = make_with_random_values<VarsType>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+
+  // Generate fluxes from the variables with random arguments
+  tuples::tagged_tuple_from_typelist<argument_tags> args{};
+  tmpl::for_each<argument_tags>(
+      [&args, &generator, &dist, &used_for_size](auto tag_v) {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        get<tag>(args) = make_with_random_values<typename tag::type>(
+            make_not_null(&generator), make_not_null(&dist), used_for_size);
+      });
+  const auto expected_fluxes = tuples::apply(
+      [&vars, &fluxes_computer](const auto&... expanded_args) {
+        return ::elliptic::first_order_fluxes<volume_dim, primal_fields,
+                                              auxiliary_fields>(
+            vars, fluxes_computer, expanded_args...);
+      },
+      args);
+
+  // Create a DataBox
+  auto box = tuples::apply(
+      [&vars, &used_for_size](const auto&... expanded_args) {
+        return db::create<tmpl::append<db::AddSimpleTags<vars_tag, fluxes_tag>,
+                                       argument_tags>>(
+            vars,
+            make_with_value<typename fluxes_tag::type>(
+                used_for_size, std::numeric_limits<double>::signaling_NaN()),
+            expanded_args...);
+      },
+      args);
+
+  // Apply the fluxes computer to the DataBox
+  tuples::tagged_tuple_from_typelist<typename VarsType::tags_list> vars_tuple{};
+  tmpl::for_each<typename VarsType::tags_list>(
+      [&vars, &vars_tuple](auto tag_v) {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        get<tag>(vars_tuple) = get<tag>(vars);
+      });
+  tuples::apply<auxiliary_fields>(
+      [&fluxes_computer, &box](const auto&... auxiliary_vars) {
+        db::mutate_apply<
+            db::wrap_tags_in<::Tags::Flux, primal_fields,
+                             tmpl::size_t<volume_dim>, Frame::Inertial>,
+            argument_tags>(fluxes_computer, make_not_null(&box),
```

Comment:
Can you clarify the design of the `fluxes_computer` object? It seems on the surface surprising that it should take as 'extra' arguments things that would typically be found in the databox when used in a `mutate_apply`. (not a change request, just trying to understand the code)

Reply:
The fluxes computer only takes "background" quantities directly out of the DataBox, i.e. those that are independent of the dynamic variables. The variables are passed in as extra args because only at the call site is it clear which variables we want to compute the fluxes of. Also the return-by-ref buffers are usually passed in at the call site to specify where to write the fluxes. As a specific example, the Poisson fluxes don't specify `Poisson::Field` in their `argument_tags` because it's not always the `Poisson::Field` tag in the DataBox that we want to apply the fluxes to. Instead, in a single elliptic solve it's typically applied to `Poisson::Field` and to a tag like `LinearSolver::Tags::Preconditioned<LinearSolver::Tags::Operand<Poisson::Field>>` plus variations thereof.

That said, I simplified the code in this test a bit, because in this case it can actually take the variables directly out of the DataBox instead of the extra indirection of copying them into a tuple.